SteamKit2/SteamKit2/Steam/CMClient.cs

```diff
@@ -23,13 +23,18 @@ namespace SteamKit2.Internal
     /// <summary>
     /// This base client handles the underlying connection to a CM server. This class should not be use directly, but through the <see cref="SteamClient"/> class.
     /// </summary>
-    public abstract class CMClient
+    public abstract class CMClient : ILogContext
     {
         /// <summary>
         /// The configuration for this client.
         /// </summary>
         public SteamConfiguration Configuration { get; }
 
+        /// <summary>
+        /// A unique identifier for this client instance.
+        /// </summary>
+        public string ID { get; }
```

Comment:
All fine but a setter would indeed be nice, see my reasoning in https://github.com/SteamRE/SteamKit/issues/503#issuecomment-549124002

Reply:
See above.